pkg/registries/adapters/helm_adapter.go

```diff
@@ -0,0 +1,322 @@
+	Engine string `protobuf:"bytes,8,opt,name=engine" json:"engine,omitempty"`
+	// The URL to an icon file.
+	Icon string `protobuf:"bytes,9,opt,name=icon" json:"icon,omitempty"`
+	// The API Version of this chart.
+	APIVersion string `protobuf:"bytes,10,opt,name=apiVersion" json:"apiVersion,omitempty"`
+	// The condition to check to enable chart
+	Condition string `protobuf:"bytes,11,opt,name=condition" json:"condition,omitempty"`
+	// The tags to check to enable chart
+	Tags string `protobuf:"bytes,12,opt,name=tags" json:"tags,omitempty"`
+	// The version of the application enclosed inside of this chart.
+	AppVersion string `protobuf:"bytes,13,opt,name=appVersion" json:"appVersion,omitempty"`
+	// Whether or not this chart is deprecated
+	Deprecated bool `protobuf:"varint,14,opt,name=deprecated" json:"deprecated,omitempty"`
+	// TillerVersion is a SemVer constraints on what version of Tiller is required.
+	// See SemVer ranges here: https://github.com/Masterminds/semver#basic-comparisons
+	TillerVersion string `protobuf:"bytes,15,opt,name=tillerVersion" json:"tillerVersion,omitempty"`
+	// Annotations are additional mappings uninterpreted by Tiller,
+	// made available for inspection by other applications.
+	Annotations map[string]string `protobuf:"bytes,16,rep,name=annotations" json:"annotations,omitempty" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+	// KubeVersion is a SemVer constraint specifying the version of Kubernetes required.
+	KubeVersion string    `protobuf:"bytes,17,opt,name=kubeVersion" json:"kubeVersion,omitempty"`
+	URLs        []string  `json:"urls"`
+	Created     time.Time `json:"created,omitempty"`
+	Removed     bool      `json:"removed,omitempty"`
+	Digest      string    `json:"digest,omitempty"`
+}
+
+// HelmAdapter - Helm Registry Adapter
+type HelmAdapter struct {
+	Config Configuration
+	Log    *logging.Logger
+	Charts map[string]ChartVersions
+}
+
+// RegistryName - Retrieve the registry name
+func (r *HelmAdapter) RegistryName() string {
+	return helmName
+}
+
+// GetImageNames - retrieve the images
+func (r *HelmAdapter) GetImageNames() ([]string, error) {
+	var imageNames []string
+
+	r.Charts = map[string]ChartVersions{}
+
+	index, err := r.getHelmIndex()
+	if err != nil {
+		return imageNames, err
+	}
+
+	for name, entry := range index.Entries {
+		if len(entry) == 0 {
+			continue
+		}
+
+		r.Charts[name] = entry
+		imageNames = append(imageNames, name)
+	}
+
+	return imageNames, nil
+}
+
+// FetchSpecs - retrieve the spec for the image names.
+func (r *HelmAdapter) FetchSpecs(imageNames []string) ([]*apb.Spec, error) {
+	var specs []*apb.Spec
+
+	for _, name := range imageNames {
+		var chartVersions []string
+		charts, ok := r.Charts[name]
+		if !ok {
+			continue
+		}
+
+		for _, chart := range charts {
+			chartVersions = append(chartVersions, chart.Version)
+		}
+		// Use the latest chart for creating the bundle
+		chart := charts[0]
```

Comment:
Might want to make sure that charts is not empty

Reply:
I am struggling to see what more to do beyond what's in `GetImageNames()`. If there are no charts, we don't add it to our list.

```
if len(entry) == 0 {
    continue
}
```